cmd/entrypoint/secrets.go

```diff
@@ -313,6 +323,67 @@ func ReconcileSecrets(ctx context.Context, sh *SnapshotHolder) error {
 	return nil
 }
 
+// Returns secretName, secretNamespace from a provided (unstructured) filter if it contains a secret
+// Returns empty strings when the secret name and/or namespace could not be found
+func FindFilterSecret(filter *unstructured.Unstructured, action func(snapshotTypes.SecretRef)) error {
+	// Just making extra sure this is actually a Filter
+	if filter.GetKind() != "Filter" && filter.GetKind() != "Filters" {
+		return errors.New("Non-Filter object in Snapshot.Filters")
+	}
+	// Only Oauth2 Filters have secrets, although they dont need to have them.
+	// This is overly contrived because Filters are unstructured to emissary since we dont have the type definitions
+	// Yes this is disgusting. It is what it is...
+	filterContents := filter.UnstructuredContent()
+	filterSpec := filterContents["spec"]
+	if filterSpec != nil {
+		mapOauth, success := filterSpec.(map[string]interface{})
+		// We need to check if all these type assertions fail since we shouldnt rely on CRD validation to protect us from a panic state
+		// I cant imagine a scenario where this would realisticly happen, but we generate a unique log message for tracability and skip processing it
+		if !success {
+			return errors.New("Filter object detected with a bogus \"spec\" field")
+		}
+		oauthFilter := mapOauth["OAuth2"]
+		if oauthFilter != nil {
+			secretName, secretNamespace := "", ""
+			// Check if we have a secretName
+			mapSecretName, success := oauthFilter.(map[string]interface{})
+			if !success {
+				return errors.New("Filter object detected with a bogus \"secretName\" field")
+			}
+			oauthSecretName := mapSecretName["secretName"]
+			if oauthSecretName != nil {
```

Comment:
no reason for switch statement and we should bail early since the logic below requires a name provided and we ignore filter if not.

```golang
if sName, ok := oauthSecretName.(string); ok  && sName != "" {
secretName = sName
} else {
// bail early since no name provided and not an error
   return nil
}
```